lib/facter/lldb.rb

```diff
@@ -0,0 +1,14 @@
+require 'facter'
+require 'json'
+
+Facter.add(:lldp) do
+  confine do
+    Facter::Util::Resolution.which('lldpctl')
+  end
+
+  setcode do
+    data = Facter::Util::Resolution.exec('lldpctl -f json')
```

Comment:
I implemented this as an external fact because I wasn't sure how long lldpctl takes to run / if it would block the puppet run. But getting rid of the systemd timer/jq dependency is also nice. Did you do some testing?

Reply:
Just did a test on a vm hosts which has 23 neighbours and runs much pretty fast

```shell
$ for i in {1..10} ; do time sudo  lldpctl -f json > /dev/null ; done                                               
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 88% cpu 0.021 total
sudo lldpctl -f json > /dev/null  0.00s user 0.02s system 88% cpu 0.023 total
sudo lldpctl -f json > /dev/null  0.01s user 0.00s system 86% cpu 0.020 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 86% cpu 0.024 total
sudo lldpctl -f json > /dev/null  0.00s user 0.01s system 88% cpu 0.019 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 90% cpu 0.020 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 86% cpu 0.020 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 85% cpu 0.016 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 86% cpu 0.019 total
sudo lldpctl -f json > /dev/null  0.01s user 0.01s system 89% cpu 0.021 total
```